tests/unit/pages/mentor-home/MentorHome.spec.js

```diff
@@ -0,0 +1,47 @@
+import { screen, render } from '@testing-library/react'
+import { MemoryRouter } from 'react-router-dom'
+import { useSelector } from 'react-redux'
+
+import { ModalProvider } from '~/context/modal-context'
+import { ConfirmationDialogProvider } from '~/context/confirm-context'
+import MentorHome from '~/pages/mentor-home/MentorHome'
+
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn()
+}))
+
+const MentorHomeWithProviders = () => (
+  <MemoryRouter>
+    <ConfirmationDialogProvider>
+      <ModalProvider>
+        <MentorHome />
+      </ModalProvider>
+    </ConfirmationDialogProvider>
+  </MemoryRouter>
+)
+
+describe('MentorHome component', () => {
+  it('should render a BecomeATutor modal when logging in for the first time', () => {
+    useSelector.mockImplementation((fn) =>
```

Comment:
test redux with Provider, and remove this mocks for useSelector

Reply:
we need to test with different isFirstLogin values

Reply:
ok